modules/dhcp/dhcp_api.rb

```diff
@@ -95,17 +131,48 @@ def load_subnet_data
     end
   end
 
-  # delete a record from a network
+  # deprecated, delete a record from a network
   delete "/:network/:record" do
     begin
       load_subnet
       load_subnet_data
 
+      logger.warn('DELETE dhcp/:network/:record endpoint has been deprecated and will be removed in future versions. '\
+                  'Please use DELETE dhcp/:network/mac/:mac_address or DELETE dhcp/:network/ip/:ip_address instead.')
+
       record = server.find_record(@subnet.network, params[:record])
-      log_halt 404, "DHCP record #{params[:network]}/#{params[:record]} not found" unless record
-      server.del_record @subnet, record
+      log_halt 404, "No DHCP record for #{params[:network]}/#{params[:record]} found" unless record
+      server.del_record(@subnet, record).to_json
+    rescue Proxy::DHCP::InvalidRecord
+      log_halt 404, "No DHCP records for #{params[:network]}/#{params[:record]} found"
+    rescue Exception => e
+      log_halt 400, e
+    end
+  end
+
+  # deletes all records for an ip address from a network
+  delete "/:network/ip/:ip_address" do
+    begin
+      load_subnet
+      load_subnet_data
+
+      server.del_records_by_ip(@subnet, params[:ip_address]).to_json
+    rescue Proxy::DHCP::InvalidRecord
+      log_halt 404, "No DHCP records for #{params[:network]}/#{params[:ip_address]} found"
```

Comment:
I don't think the message and 404 status are appropriate in this case -- the only time this exception is raised when the record is not deletable (AKA lease), and that is probably going away at some point.

I looked at a few providers -- there doesn't seem to be an agreement on how to handle deletion of missing hosts (it wasn't required before these PR, however): ms and infoblox return without an error, isc will probably return with an error, and I'm not sure re: libvirt.

I'm ok with either returning or not returning 404 to the caller, but both cases will require changes to providers which now need to correctly report (or not report) the "missing record" error. If you don't have an opportunity to test required changes for isc and libvirt, I would suggest fixing this in a follow-up PR (which I can pick up).